ekncontent/configure.ac

```diff
@@ -0,0 +1,196 @@
+# Unlike serial-tests, this option is accepted by Automake 1.11
+# tar-ustar is required because there may be files whose entire paths exceed
+# 99 characters.
+# subdir-objects is for forward compatibility with Automake 2.0 and can be
+# removed when we upgrade to that version.
+AM_INIT_AUTOMAKE([-Wall -Wno-portability foreign 1.11 parallel-tests tar-ustar subdir-objects])
+# Avoid spewing garbage over the terminal ('make V=1' to see the garbage)
+AM_SILENT_RULES([yes])
+# Initialize Libtool; don't build static libraries
+LT_INIT([disable-static])
+# Keep Autotools macros local to this source tree
+AC_CONFIG_MACRO_DIR([m4])
+
+AC_CACHE_SAVE
+
+# Variables to define
+# -------------------
+# Version numbers and macros
+EKN_CONTENT_API_VERSION=_EKN_CONTENT_API_VERSION_MACRO
+AC_SUBST(EKN_CONTENT_API_VERSION)
+EKN_CONTENT_API_NAME="eos-knowledge-content-$EKN_CONTENT_API_VERSION"
+AC_SUBST(EKN_CONTENT_API_NAME)
+EKN_CONTENT_INTERFACE_AGE=_EKN_CONTENT_INTERFACE_AGE_MACRO
+AC_SUBST([EKN_CONTENT_MAJOR_VERSION], [_EKN_CONTENT_API_VERSION_MACRO])
+AC_SUBST([EKN_CONTENT_MINOR_VERSION], [_EKN_CONTENT_MINOR_VERSION_MACRO])
+AC_SUBST([EKN_CONTENT_MICRO_VERSION], [_EKN_CONTENT_MICRO_VERSION_MACRO])
+AC_DEFINE([EKN_CONTENT_VERSION], [_EKN_CONTENT_VERSION_MACRO], [Full version number])
+
+# Shared library versioning; calculated automatically.
+# See http://www.gnu.org/software/libtool/manual/html_node/Updating-version-info.html
+EKN_CONTENT_LT_CURRENT=m4_eval(100 * _EKN_CONTENT_MINOR_VERSION_MACRO + _EKN_CONTENT_MICRO_VERSION_MACRO - _EKN_CONTENT_INTERFACE_AGE_MACRO)
+EKN_CONTENT_LT_REVISION=$EKN_CONTENT_INTERFACE_AGE
+EKN_CONTENT_LT_AGE=$EKN_CONTENT_LT_CURRENT
+EKN_CONTENT_LT_VERSION_INFO="$EKN_CONTENT_LT_CURRENT:$EKN_CONTENT_LT_REVISION:$EKN_CONTENT_LT_AGE"
+AC_SUBST(EKN_CONTENT_LT_VERSION_INFO)
+
+# Required build tools
+# --------------------
+# Make sure we can create directory hierarchies
+AC_PROG_MKDIR_P
+# Make sure we can symlink
+AC_PROG_LN_S
+AS_IF([test "$LN_S" = "ln -s"], [], [AC_MSG_ERROR([Symlink support required.])])
+# C compiler
+AC_PROG_CC
+AC_PROG_CC_C99
+# Make sure the C compiler supports per-target CFLAGS
+AM_PROG_CC_C_O
+# Library configuration tool
+PKG_PROG_PKG_CONFIG
+# Gtk-doc; 0.18 required for Markdown parsing
+GTK_DOC_CHECK([1.18], [--flavour no-tmpl])
+# GObject Introspection
+GOBJECT_INTROSPECTION_REQUIRE([1.30])
+# Various tools
+AC_PROG_AWK  # needed for TAP driver
+AC_PROG_SED  # needed for finding files for coverage
+AC_REQUIRE_AUX_FILE([tap-driver.sh])
+AC_PATH_PROG([GIO_QUERYMODULES], [gio-querymodules], [notfound])
+AS_IF([test x$GIO_QUERYMODULES = xnotfound], [AC_MSG_ERROR([gio-querymodules not found.
+Try installing "glib" package.])])
+
+AC_CACHE_SAVE
+
+# Check that the compiler really supports C99
+AS_IF([test "x$ac_cv_prog_cc_c99" = "xno"], [AC_MSG_ERROR([C99 is required.])])
+
+# Configure options
+# -----------------
+# --enable-strict-flags: Compile with strict compiler flags. Done automatically
```

Comment:
I would get rid of this whole option and just use the default settings of [`AX_COMPILER_FLAGS`](https://www.gnu.org/software/autoconf-archive/ax_compiler_flags.html)